### PR TITLE
ROC-3442: Loosen the CMC reference number check criteria

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-3442

### Change description

If the reference number is not in `\d\d\dMC\d\d\d` format then redirect to MCOL, even if provided reference number is blank.

### Work checklist

- [X] Unit tests added where applicable

### Developer self-QA run statement

- [X] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/common/utils/isNonCMCReference.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ *
+ * @param {string} referenceNumber the reference number
+ * @returns {boolean} true if the number if not a CMC reference number, false otherwise
+ */
+export function isNonCMCReference (referenceNumber?: string): boolean {
+  if (!referenceNumber) {
+    return true
+  }
+
+  return !referenceNumber.match(/^\d\d\dMC\d\d\d$/)
```
Comment (timja-kainos) [inline src/main/common/utils/isNonCMCReference.ts]: this doesn't look to be case insensitive?

File: src/main/common/utils/isNonCMCReference.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Checks if given reference number is not a CMC reference number.
+ *
+ * @param {string} referenceNumber the reference number
+ * @returns {boolean} true if the number if not a CMC reference number, false otherwise
+ */
+export function isNonCMCReference (referenceNumber?: string): boolean {
```
Comment (timja-kainos) [inline src/main/common/utils/isNonCMCReference.ts]: not hugely sure if this function is really needed, there's already a validator for the format, the only reason not to just use the validator would be it handles it being empty,

Consider refactoring this but 👍

Comment (rlewan) [reply src/main/common/utils/isNonCMCReference.ts]: Refactored by removing `@Matches` from `ClaimReference` model.

**Comment by rlewan** [comment]
Green build on dev Jenkins: https://dev-build.reform.hmcts.net/blue/organizations/jenkins/hmcts%2Fcmc-citizen-frontend/detail/PR-495/4/pipeline/

**Pull request merged**